python/semantic_kernel/kernel.py

```diff
@@ -828,7 +829,7 @@ def import_native_skill_from_directory(
         return {}
 
     def import_semantic_skill_from_directory(
-        self, parent_directory: str, skill_directory_name: str
+        self, parent_directory: str, skill_directory_name: str, encoding=locale.getencoding()
```

Comment:
The `locale.getencoding()` will work with Python 3.11+. To support Python 3.8 - 3.10, we'd need to use something like:

```
import locale
import sys

if sys.version_info >= (3, 11):
    default_encoding = locale.getencoding()
else:
    default_encoding = locale.getpreferredencoding()
```

This could be represented in the method definition as

```
encoding=locale.getpreferredencoding() if sys.version_info < (3, 11) else locale.getencoding()
```